### ISSUE TITLE
Unable to set different locales for calendar and date picker

### ISSUE THREAD

**Issue by iam-sreekanth** [schedule-x/schedule-x#817]
I am trying to configure my calendar to use the `en-US` locale while setting the date picker to `en-GB`. My goal is to display dates in the `dd/MM/yyyy` format. However, despite setting `locale: 'en-GB'` for the date picker, the format remains `MM/dd/yyyy`.

Steps to Reproduce:
1. Configure the calendar with `locale: 'en-US'`.
2. Configure the date picker with `locale: 'en-GB'`.
3. Select a date in the date picker.
4. Observe that the format is not in `dd/MM/yyyy`.

The selected date in the date picker should be displayed in `dd/MM/yyyy` format.

The selected date is displayed in `MM/dd/yyyy` format, which is not the desired behavior.

locale: 'en-US',
datePicker: {
    locale: 'en-GB',
}

While I found a partial solution, it introduced new issues:

Setting the calendar's locale to en-GB makes the date picker show the date in dd/MM/yyyy format (as desired).
However, the time axis and event grid also switch to 24-hour format, omitting AM/PM.

This behavior makes it impossible to achieve the desired combination:

Dates: dd/MM/yyyy format.
Times: 12-hour format with AM/PM in the time axis and event grid.

**Environment**

1. Package Version:  "@schedule-x/react": "^2.9.0"
2. TypeScript Version:  "typescript": "^5.5.3"
3. Framework: "react": "^18.3.1"

**Comment by tomosterlund** [comment]
What would be your reasoning for supporting a mixture of locales and date-time formats?

To me it would seem like a confusing feature to display UK date formatting to people using US English, and similarly other way around. If you just want to keep _US-like_ English but another date format, you could also consider adding translations for Canadian. Their date format is YYYY-MM-DD.

**Comment by iam-sreekanth** [comment]
I understand your point, and I appreciate the feedback. However, there are a few reasons why supporting a mixture of locales and date-time formats can be important in some cases:

1. User Preferences and Regional Standards: In many global applications, users may be accustomed to a specific date format regardless of the locale. For example, while the language and time format might default to en-US (with AM/PM and 12-hour clock), certain users (especially those in regions like Europe) may prefer the dd/MM/yyyy date format for clarity and consistency in date representation. It's crucial to offer flexibility in these cases without forcing a full locale switch, especially when cultural differences in date formatting are important for accurate interpretation.

2. Internationalization & Consistency: Different regions have distinct date formatting conventions, and it's not always necessary to adhere to a strict one-to-one relationship between language and format. Allowing users to mix formats, where the calendar date follows a different standard (e.g., dd/MM/yyyy in UK format) while keeping the time format and language consistent with US English, ensures clarity and usability for a wider audience.

3. Practicality in Applications: For certain industries, such as those dealing with international clients, using a globally understood date format like dd/MM/yyyy while maintaining the US English locale is common. This avoids confusion, especially when sharing dates across regions (e.g., a US-based application catering to UK clients).

4. Example of Canadian Locale: While Canadian English does use YYYY-MM-DD as the standard date format, it’s not the most commonly used format for all users. For example, many Canadian users may still prefer to see dates in dd/MM/yyyy format for easier reading, even if the locale remains en-CA.

In conclusion, supporting this feature provides flexibility for users who need it without confusing the underlying locale for language and time formatting. It offers the best of both worlds: localized language and time formatting with a user-friendly date format.
